cloud/grouping.py

```diff
@@ -0,0 +1,134 @@
+# Copyright (c) 2015 Mirantis Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the License);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an AS IS BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied.
+# See the License for the specific language governing permissions and#
+# limitations under the License.
+
+
+import yaml
+
+import cloud
+
+from cloudferrylib.os.identity import keystone
+from cloudferrylib.os.network import neutron
+from cloudferrylib.os.compute import nova_compute
+
+
+class Grouping(object):
+    def __init__(self, config, group_file, cloud_id):
+        self.config = config
+        self.group_config = self.get_group_conf(group_file)
+        resources = {'identity': keystone.KeystoneIdentity,
+                     'network': neutron.NeutronNetwork,
+                     'compute': nova_compute.NovaCompute}
+        self.cloud = cloud.Cloud(resources, cloud_id, config)
+
+        self.network = self.cloud.resources['network']
+        self.compute = self.cloud.resources['compute']
+        self.identity = self.cloud.resources['identity']
+
+        self.groups = {}
+
+    @staticmethod
+    def get_group_conf(group_file):
+        with open(group_file) as gf:
+            group_conf = yaml.load(gf)
+        return group_conf
+
+    @staticmethod
+    def write_groups(file_name, groups):
+        with open(file_name, 'w') as gr:
+            yaml.dump(groups, gr)
+
+    def group(self):
+        group_by = self.group_config.pop('group_by')
+
+        step = 0
+
+        for grouping in group_by:
+            groups = self.group_by(grouping, step)
+            if not step:
+                self.groups = groups
+            step += 1
+
+        self.make_users_group(self.groups)
+
+        self.write_groups(self.config.migrate.group_file_path, self.groups)
+
+    def group_by(self, target, step):
+        group_rules_map = {'tenant': self.group_nested_tenant,
+                           'network': self.group_nested_network,
+                           }
+        try:
```

Comment:
I believe we don't need try-except case in this situation
`group_func = group_rules_map.get(target)
if not group_func:
    raise RuntimeError("message")`